### PR TITLE
media-tv/plex-media-server: readd hardened support, initscript

### PR DESCRIPTION
**Pull request by joecool1029**
These changes will bring us in line with plex-overlay.

- start_pms is hacky and removed, we shouldn't ship it.
- the service
- we had pax marking, someone removed it, but it was requested before so readding.
- openrc and systemd support
- Lift mirror restriction (again): https://bugs.gentoo.org/600696

Signed-off-by: Joe Kappus <joe@wt.gd>

### CODE REVIEW

File: media-tv/plex-media-server/plex-media-server-1.30.2.6563-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+KEYWORDS="-* ~amd64 ~arm64 ~x86"
+RESTRICT="bindist"
+
+DEPEND="
+	acct-group/plex
+	acct-user/plex"
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"${FILESDIR}/${PN}.service.patch" )
```
Comment (thesamesam) [inline media-tv/plex-media-server/plex-media-server-1.30.2.6563-r1.ebuild:30]:
```suggestion
	"${FILESDIR}/${PN}.service.patch"
)
```

File: media-tv/plex-media-server/plex-media-server-1.30.2.6563-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+	# Install the OpenRC init/conf files
+	newinitd "${FILESDIR}/${PN}.init.d" ${PN}
+	newconfd "${FILESDIR}/${PN}.conf.d" ${PN}
+
+	# Install systemd service file
+	systemd_newunit "${ED}"/usr/lib/plexmediaserver/lib/plexmediaserver.service "${PN}.service"
+
+	# Add pax markings to some binaries so that they work on hardened setup
+	for f in "${BINS_TO_PAX_MARK[@]}"; do
```
Comment (thesamesam) [inline media-tv/plex-media-server/plex-media-server-1.30.2.6563-r1.ebuild:67]:
```suggestion
	local f
	for f in "${BINS_TO_PAX_MARK[@]}"; do
```

File: media-tv/plex-media-server/plex-media-server-1.30.2.6563-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+MY_PV="${PV}-3d4dc0cce"
+MY_URI="https://downloads.plex.tv/plex-media-server-new"
+
+DESCRIPTION="Free media library that is intended for use with a plex client"
+HOMEPAGE="https://www.plex.tv/"
+SRC_URI="
+	amd64? ( ${MY_URI}/${MY_PV}/debian/plexmediaserver_${MY_PV}_amd64.deb )
+	arm64? ( ${MY_URI}/${MY_PV}/debian/plexmediaserver_${MY_PV}_arm64.deb )
+	x86? ( ${MY_URI}/${MY_PV}/debian/plexmediaserver_${MY_PV}_i386.deb )"
```
Comment (thesamesam) [inline media-tv/plex-media-server/plex-media-server-1.30.2.6563-r1.ebuild:16]:
```suggestion
	x86? ( ${MY_URI}/${MY_PV}/debian/plexmediaserver_${MY_PV}_i386.deb )
"
```

**Comment by joecool1029** [comment]
@thesamesam can you merge this? I want to get the tree caught up and while I've tested the revbump on my non-overlay systems, there's a small possibility it could cause issues for users that didn't use the default plex user permissions. Honestly amazed more bugs weren't open for the buggy start_pms script over the years.

**Pull request closed without merging**